src/fireedge/src/modules/containers/Providers/Providers.js

```diff
@@ -0,0 +1,205 @@
+      hide()
+      await deleteProvider({ id })
+    } catch {}
+  }
+
+  useEffect(() => {
+    successDelete && enqueueSuccess(T.SuccessProviderDeleted, deletedProviderId)
+  }, [successDelete])
+
+  return (
+    <TranslateProvider>
+      <>
+        <ListHeader
+          title={T.Providers}
+          reloadButtonProps={{
+            'data-cy': 'refresh-provider-list',
+            onClick: () => refetch(),
+            isSubmitting: isFetching,
+          }}
+          addButtonProps={{
+            'data-cy': 'create-provider',
+            onClick: () => history.push(PATH.INFRASTRUCTURE.PROVIDERS.CREATE),
+          }}
+          searchProps={{ handleChange }}
+        />
+        <Box p={3}>
+          {error ? (
+            <AlertError>{T.CannotConnectOneProvision}</AlertError>
+          ) : (
+            <ListCards
+              list={result ?? providers}
+              gridProps={{ 'data-cy': 'providers' }}
+              CardComponent={ProvisionCard}
+              cardsProps={({ value: { ID, NAME, TEMPLATE } }) => ({
+                image: providerConfig[TEMPLATE?.PLAIN?.provider]?.image,
+                isProvider: true,
+                handleClick: () => show({ id: ID, title: `#${ID} ${NAME}` }),
+                actions: [
+                  {
+                    handleClick: () =>
+                      history.push(
+                        generatePath(PATH.INFRASTRUCTURE.PROVIDERS.EDIT, {
+                          id: ID,
+                        })
+                      ),
+                    icon: <EditIcon />,
+                    cy: 'provider-edit',
+                  },
+                  {
+                    handleClick: () =>
+                      show({
+                        id: ID,
+                        delete: true,
+                        title: (
+                          <Translate
+                            word={T.DeleteSomething}
+                            values={`#${ID} ${NAME}`}
+                          />
+                        ),
+                        handleAccept: () => handleDelete(ID),
+                      }),
+                    icon: <DeleteIcon />,
+                    isSubmitting: isDeleting,
+                    color: 'error',
+                    cy: 'provider-delete',
+                  },
+                ],
+              })}
+            />
+          )}
+        </Box>
+        {display &&
+          dialogProps?.id &&
+          (!dialogProps?.delete ? (
+            <DialogProvider
+              hide={hide}
+              id={dialogProps.id}
+              dialogProps={dialogProps}
+            />
+          ) : (
+            <DialogConfirmation handleCancel={hide} {...dialogProps}>
+              <p>
+                <Translate word={T.DoYouWantProceed} />
+              </p>
+            </DialogConfirmation>
+          ))}
+      </>
+    </TranslateProvider>
+  )
+}
+
+const DialogProvider = memo(
+  ({ id, hide, dialogProps }) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).